tests/components/eq3btsmart/test_config_flow.py

```diff
@@ -0,0 +1,150 @@
+"""Test the eq3btsmart config flow."""
+
+from unittest.mock import patch
+
+from homeassistant import config_entries
+from homeassistant.components.bluetooth import BluetoothServiceInfoBleak
+from homeassistant.components.eq3btsmart.const import DOMAIN
+from homeassistant.const import CONF_MAC, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import MAC, NAME
+
+from tests.common import MockConfigEntry
+
+
+async def test_user_flow(hass: HomeAssistant) -> None:
+    """Test we can handle a regular successflow setup flow."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.eq3btsmart.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_MAC: MAC, CONF_NAME: NAME},
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == NAME
+    assert result["data"] == {
+        CONF_MAC: MAC,
+        CONF_NAME: NAME,
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_user_flow_invalid_mac(hass: HomeAssistant) -> None:
+    """Test we handle invalid mac address."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.eq3btsmart.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_MAC: "invalid", CONF_NAME: NAME},
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {CONF_MAC: "invalid_mac_address"}
+    assert len(mock_setup_entry.mock_calls) == 0
```

Comment:
Please finish all config flow tests to reach CREATE_ENTRY or ABORT states to ensure they can recover from failures